File: apps/go/domain/type_/fizz_buzz_type_03.go
package type_

import (
	"strconv"

	"github.com/k2works/getting-started-tdd/apps/go/domain/model"
)

// FizzBuzzType03 は FizzBuzz のみ返し、それ以外は数値を返します。
type FizzBuzzType03 struct{ fizzBuzzTypeBase }

func (f FizzBuzzType03) Generate(number int) model.FizzBuzzValue {
	if f.isFizz(number) && f.isBuzz(number) {
		return model.NewFizzBuzzValue(number, "FizzBuzz")
	}
	return model.NewFizzBuzzValue(number, strconv.Itoa(number))
}

File: apps/java/src/main/java/tdd/fizzbuzz/FizzBuzzType03.java
package tdd.fizzbuzz;

public class FizzBuzzType03 extends FizzBuzzType {

    @Override
    public String generate(int number) {
        if (isFizzBuzz(number)) {
            return "FizzBuzz";
        }
        return Integer.toString(number);
    }
}
